### PR TITLE
disable recent job results list for sparc and improve test

### PR DESCRIPTION
**Pull request by franku**
- rename test "recently-used-jobs" to "statefile"
- disable all "statefile" tests for sparc
- use temporary state files instead of original files in the source code which may be deleted during the test in case of an error

#### Checklist
- [x] [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [x] [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [x] [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

#### Add some information

- [x] Add a small description to the CHANGELOG.md file and refer to your PR using this syntax '[PR #xyz]'
- [x] Add your name to the AUTHORS file

#### Keep spirit!
- [x] Do not be afraid to hand in a PR!

### CODE REVIEW

File: core/src/tests/recent_job_results_list.cc [hunk 1]
```diff
@@ -48,20 +48,50 @@ static bool StateFileExists(const char* dir, const char* name, int port)
   return stat(path.c_str(), &buf) == 0;
 }
 
+static bool CopyTestStateFileFromOriginal(std::string orig_path,
```
Comment (pstorz) [inline core/src/tests/recent_job_results_list.cc:51]: As we now have c++17 available this should be easier with 
std::filesystem::copy

**Review by pstorz — CHANGES_REQUESTED** [review]
I would suggest to change the name of the tests:

recent_job_results_list -> statefile:
read_job_results_from_flie -> read
write_job_results_to_file -> write
read_job_results_from_file_truncated_jobs -> handle_truncated_jobs
read_job_results_from_file_truncated_header -> handle_truncated_headers
read_job_results_from_file_not_exist -> handle_empty_file

File: core/src/tests/recent_job_results_list.cc (hunk 1 above)
Comment (arogge) [reply core/src/tests/recent_job_results_list.cc:51]: the filesystem interface is not really usable today. A lot of platforms require to link with a special lib and some don't support the interface yet.

File: core/src/tests/CMakeLists.txt [hunk 2]
```diff
@@ -264,14 +266,29 @@ bareos_add_test(
...
-set_tests_properties(
-  gtest:recent_job_results_list.read_job_results_from_file PROPERTIES LABELS
-                                                                      "broken"
-)
-set_tests_properties(
-  gtest:recent_job_results_list.write_job_results_to_file PROPERTIES LABELS
-                                                                     "broken"
-)
+
+if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "sparc")
```
Comment (arogge) [inline core/src/tests/CMakeLists.txt:270]: it doesn't make sense to check for the platform we're building on, we should check the platform we're building for.
```suggestion
if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "sparc")
```
Nevertheless, it should be rather simple to provide test-data for big endian, too. So there's no real point in disabling it.

Comment (franku) [reply core/src/tests/CMakeLists.txt]: if(HAVE_BIG_ENDIAN) ..

**Review by pstorz — APPROVED** [review]
Looks good.

**Pull request merged**